shared/lib_irradproc.cpp

```diff
@@ -34,2465 +34,3663 @@ OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+        timeStepSunPosition[2] = 2;
+    }
+    // timestep encompasses the sunset
+    else if (delt > 0 && t_cur > t_sunset - delt / 2.0 && t_cur <= t_sunset + delt / 2.0)
+    {
+        double t_calc = ((t_cur - delt / 2.0) + t_sunset) / 2.0; // midpoint of beginning of timestep and sunset
+        int hr_calc = (int)t_calc;
+        double min_calc = (t_calc - hr_calc) * 60.0;
+
+        timeStepSunPosition[0] = hr_calc;
+        timeStepSunPosition[1] = (int)min_calc;
+
+        solarpos_spa(year, month, day, hr_calc, min_calc, 0.0, latitudeDegrees, longitudeDegrees, timezone, dut1, elevation, pressure, temp, tiltDegrees, surfaceAzimuthDegrees, sunAnglesRadians);
+
+        timeStepSunPosition[2] = 3;
+    }
+    // timestep is not sunrise nor sunset, but sun is up  (calculate position at provided t_cur)
+    else if ((t_sunrise < t_sunset && t_cur >= t_sunrise && t_cur <= t_sunset) || //this captures normal daylight cases
+        (t_sunrise > t_sunset && (t_cur <= t_sunset || t_cur >= t_sunrise))) //this captures cases where sunset (from previous day) is 1:30AM, sunrise 2:30AM, in arctic circle
+    {
+        timeStepSunPosition[0] = hour;
+        timeStepSunPosition[1] = (int)minute;
+        solarpos_spa(year, month, day, hour, minute, 0.0, latitudeDegrees, longitudeDegrees, timezone, dut1, elevation, pressure, temp, tiltDegrees, surfaceAzimuthDegrees, sunAnglesRadians);
+        timeStepSunPosition[2] = 1;
+    }
+    else
+    {
+        // sun is down, assign sundown values
+        solarpos_spa(year, month, day, hour, minute, 0.0, latitudeDegrees, longitudeDegrees, timezone, dut1, elevation, pressure, temp, tiltDegrees, surfaceAzimuthDegrees, sunAnglesRadians);
+        timeStepSunPosition[0] = hour;
+        timeStepSunPosition[1] = (int)minute;
+        timeStepSunPosition[2] = 0;
+    }
+
+
+    planeOfArrayIrradianceFront[0] = planeOfArrayIrradianceFront[1] = planeOfArrayIrradianceFront[2] = 0;
+    diffuseIrradianceFront[0] = diffuseIrradianceFront[1] = diffuseIrradianceFront[2] = 0;
+    surfaceAnglesRadians[0] = surfaceAnglesRadians[1] = surfaceAnglesRadians[2] = surfaceAnglesRadians[3] = surfaceAnglesRadians[4] = 0;
+
+    // do irradiance calculations if sun is up
+    if (timeStepSunPosition[2] > 0)
+    {
+        // compute incidence angles onto fixed or tracking surface
+        incidence(trackingMode, tiltDegrees, surfaceAzimuthDegrees, rotationLimitDegrees, sunAnglesRadians[1], sunAnglesRadians[0],
+            enableBacktrack, groundCoverageRatio, forceToStow, stowAngleDegrees, surfaceAnglesRadians);
+        if (radiationMode < irrad::POA_R) {
+            double hextra = sunAnglesRadians[8];
+            double hbeam = directNormal * cos(sunAnglesRadians[1]); // calculated beam on horizontal surface: sunAnglesRadians[1]=zenith
+            if (directNormal < 0)
+            {
+                hbeam = 0;
+            }
+            /*if (directNormal < 0 && radiationMode == irrad::GH_DF)
```

Comment:
why is this commented out?

Reply:
That portion was from testing for edge cases and forgot to remove it. The code above it works in its place